# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/client/TextureClientSharedSurface.cpp
# Commit: dd00e2da3a0f
# Full Hash: dd00e2da3a0f11a7cc0af9e5ece21a339f4393eb
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 04:04:33
# Regressor Bug: 1632249
# File Overlap Count: 3
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 69cc1de6f754 -r dd00e2da3a0f gfx/layers/client/TextureClientSharedSurface.cpp
--- a/gfx/layers/client/TextureClientSharedSurface.cpp	Wed Jun 10 21:57:19 2020 +0000
+++ b/gfx/layers/client/TextureClientSharedSurface.cpp	Wed Jun 10 22:21:02 2020 +0000
@@ -20,16 +20,17 @@
 namespace layers {
 
 SharedSurfaceTextureData::SharedSurfaceTextureData(
-    UniquePtr<gl::SharedSurface> surf)
-    : mSurf(std::move(surf)) {}
+    const SurfaceDescriptor& desc, const gfx::SurfaceFormat format,
+    const gfx::IntSize size)
+    : mDesc(desc), mFormat(format), mSize(size) {}
 
 SharedSurfaceTextureData::~SharedSurfaceTextureData() = default;
 
 void SharedSurfaceTextureData::Deallocate(LayersIPCChannel*) {}
 
 void SharedSurfaceTextureData::FillInfo(TextureData::Info& aInfo) const {
-  aInfo.size = mSurf->mSize;
-  aInfo.format = gfx::SurfaceFormat::UNKNOWN;
+  aInfo.size = mSize;
+  aInfo.format = mFormat;
   aInfo.hasIntermediateBuffer = false;
   aInfo.hasSynchronization = false;
   aInfo.supportsMoz2D = false;
@@ -37,26 +38,22 @@
 }
 
 bool SharedSurfaceTextureData::Serialize(SurfaceDescriptor& aOutDescriptor) {
-  return mSurf->ToSurfaceDescriptor(&aOutDescriptor);
+  aOutDescriptor = mDesc;
+  return true;
+}
+/*
+static TextureFlags FlagsFrom(const SharedSurfaceDescriptor& desc) {
+  auto flags = TextureFlags::ORIGIN_BOTTOM_LEFT;
+  if (!desc.isPremultAlpha) {
+    flags |= TextureFlags::NON_PREMULTIPLIED;
+  }
+  return flags;
 }
 
 SharedSurfaceTextureClient::SharedSurfaceTextureClient(
-    SharedSurfaceTextureData* aData, TextureFlags aFlags,
-    LayersIPCChannel* aAllocator)
-    : TextureClient(aData, aFlags, aAllocator) {
-  mWorkaroundAnnoyingSharedSurfaceLifetimeIssues = true;
-}
-
-already_AddRefed<SharedSurfaceTextureClient> SharedSurfaceTextureClient::Create(
-    UniquePtr<gl::SharedSurface> surf, gl::SurfaceFactory* factory,
-    LayersIPCChannel* aAllocator, TextureFlags aFlags) {
-  if (!surf) {
-    return nullptr;
-  }
-  TextureFlags flags = aFlags | TextureFlags::RECYCLE | surf->GetTextureFlags();
-  SharedSurfaceTextureData* data =
-      new SharedSurfaceTextureData(std::move(surf));
-  return MakeAndAddRef<SharedSurfaceTextureClient>(data, flags, aAllocator);
+    const SharedSurfaceDescriptor& aDesc, LayersIPCChannel* aAllocator)
+    : TextureClient(new SharedSurfaceTextureData(desc), FlagsFrom(desc),
+aAllocator) { mWorkaroundAnnoyingSharedSurfaceLifetimeIssues = true;
 }
 
 SharedSurfaceTextureClient::~SharedSurfaceTextureClient() {
@@ -80,6 +77,7 @@
     delete data;
   }
 }
+*/
 
 }  // namespace layers
 }  // namespace mozilla